### DISCUSSION TITLE
User Discussion Area

### DISCUSSION THREAD

**Discussion by huss** [OpenEnergyDashboard/OED#556] [Ideas]
You can propose new ideas, ask about usage and other areas related to those using OED.

**Comment by huss** [comment]
### Potential future features of the Open Energy Dashboard

A wide range of institutions and people have suggested features for OED. The following list of such requests is somewhat unordered. If there is an issue or progress on the item then it is noted. We don't know when we will implement them and input from users does impact what we do sooner rather than later. If you want to make a comment on a potential feature, support a request or suggest another feature then you can do it here. You can also directly [contact the project](https://openenergydashboard.github.io/contact.html).

- Acquire meter data from more types of meters including BACnet, Modbus, BAS systems.
  - OED will be accepting Obvius data as of verson 0.7 (issue #520). That will allow acquisition from hundreds of devices (but you do need Obvius hardware).
  - OED will soon (planned for version 0.8) have the ability to accept CSV files for meter information and meter data. (issue #370)
  - OED does not plan to do BACnet or Modbus directly but depend on other vendors to acquire that data where OED has interfaces to accept the data. There is substantial expertise needed and the exact interface varies with devices. Obvius was the first such integration.
  - OED has an Application Programming Interface (API) that allows for adding new meter types to the system. We welcome working with people to add new meter types. This is the current method we have to add more meter types if you don't want to provide data in CSV format.
- Allow arbitrary arithmetic on meter or combinations of meters data to correct for issues, scale readings, and treat generated energy (such as solar) as negative.
  - This will likely be done as part of resource generalization or after the initial implementation is done. Note that this will likely be a general scale and shift ability where negating will be a special case.
- Radar/clock type visualization of data. (issue #208)
- Normalize graphs for # people, size (sq. feet, cubic feet), weather (degree heat/cool days, sunny/cloudy, wind).
    Show energy cost for usage.
  - We are in the planning stages of modifying OED to allow for general resource types. We plan to do normalization as later parts of that process.
- Show usage in units that a typical user might be interest in, e.g., CO2, miles driven by a car, cost, etc.
  - The planned generalization of resources will include capabilities for this type of representation of the data. Thus, this should happen after the initial generalization is done.
- Allow baselining of displayed data. This would be used to remove usage when the building is empty so the net usage from occupants is show. The preference was to make its inclusion an option for a graphic.
  - OED has tested the underlying idea and shown it works. We have a crude mock-up that allows admins to select time ranges to use for baselining and OED automatically calculates the baseline value. We also verified a way to include baselines in graphics.
  - The tested system would allow for changing the baseline value so different time periods could have different values. This would support structural or mechanical changes to a building that impact the baseline.
  - To do this within a general system is related to how OED will be generalizing resource types. As such, it will planned for after that is complete unless there is stronger demand.
- To speed up OED presenting graphical data and to limit the data transferred (esp. useful on mobile devices), OED does data averaging on the server so it only returns the number of points the user can realistically display on their device. You can zoom and scroll through the data including getting to the raw meter data if you zoom to a sufficiently small time range. However, some users would like to be able to better understand the underlying meter data when looking at longer time frames. This would help in locating anomalous areas worth looking at in more detail.
  - OED version 0.8 plans to have a feature to allow the user to download the raw meter readings seen in a graphic. (issue #101) This will allow people to engage with the data outside of OED.
  - The raw data dump is useful but does not help in easily determining where issues might exist by using OED. The planned OED solution is to have a graphic that shows the averaged points (as OED currently does) but also display the range of values that are represented by the averaged point. One simply way to do this is to show the maximum and minimum value above and below the averaged point (as in a type of error bar). If these two additional value differ significantly from the average then you know the averaging has glossed over details that might be of interest. You can then zoom into this region to see more details.
  - Another step that would be farther off would be to use fancier techniques and automated ways to show details in regions with greater variation in data values.
- Have building pages.
  - OED has groups that allow for combining meter data. One use of groups is to determine usage a building. This partly addresses this issue.
  - OED has decided, for the time being, not to create unique building pages. It seems organizations have varying ideas of what those pages should look like. Creating a general system that allows customization would be substantial work. For now, OED decided organizations can use the chart links that recreate graphics to embed graphics within any site they wish, including building pages they create.
- Track projects with expected and actual savings.
- Allow arbitrary shift in time between current and previous in comparison graphics along with arbitrary time width.
  - The OED system has this ability. We have not created an interface to allow users to do this. It could be done if there was enough demand for the feature.
- Logins for people other than admins such as student, visitor, facilities.
  - OED is already implementing additional user roles to support update/download of data with its inclusion expected with version 0.8 (issue #544).
  - OED has asked people if they want to control access with types of logins. There has yet to be significant demand for this feature.
- Other graphical representations such as heat maps, low/high, comparison to typical usage, inclusion of temperature on line graphs, comparison to targets.
  - OED would like to do this in the future. When it will happen is uncertain.
- Create PDF poster for competitions from template but customizable.
- Multi-institution comparison & competition.
  - The question of sharing data and extent of interest has not been sufficiently addressed.
- Create data for standard reports including ENERGY STAR, GRITS.
- Social media to ask questions, report issues in sustainability, question of the day.
- Display peak loads and also warn if approaching a peak so can avoid going higher.
- Notifications (email, text) to admins when approaching peak, unusual readings that might indicate a problem, exceeding threshold.
  - OED already supports sending admins emails with any issues the system encounters. What would be needed is a system to determine when these situations occur and a plan for how to efficiently detect them.
- Fit data to curves.
- Track people in buildings over time to see if interventions spread and stay over time.
- Psychology for change such as cute images, energy tips.
- Allow for line graphs that compare one time frame to another. For example, compare the current year to the previous year. This is different than the comparison bar graphs that show the total usage in that range. Here you would see the points of usage across the time desired.
- Pie charts graphical representation.
  - This has come up a few times. The OED project is unclear on the value of them versus bar charts. Having said that, it is not too hard to add them to OED. If there is sufficient demand and reasons then we can do them.

**Reply by Nithish-dot** [reply]
If possible can the team upload a screen recording on how to run the project from beginning making it BEGINNER FRIENDLY, Since I'm new to git ,It is  possible to  get along with the documentation and make the project work in my local system but if its is visually available it might be even more helpful, for beginners like me and might even reach out to more people who are in need of it....thank u

**Reply by huss** [reply]
Thanks to @Nithish-dot for making this suggestion. Someone planned to do this last spring but did not get it done. It is on my TODO list and I hope to get to it at some point. OED also wants to update the directions to make them friendlier. Any thoughts are welcome. I hope your message means it will get done sooner.

**Comment by Nithish-dot** [comment]
Thank you lot for your reply... Sir... It was just a request from my side... That's it.... I'm just a starter.... In learning programming, I found GitHub projects lot more of an opportunity to learn a lot of things than YouTube's hello world programs...Hopefully waiting for your projects demo video...... 😍🥰

Nithish
________________________________

**Comment by psdavis** [comment]
Are there any plans to create a viewer user profile for sites that would like to restrict access to the dashboard?

**Comment by huss** [comment]
@psdavis Sorry for the delay in responding. Currently OED allows sites to restrict access to meters/groups via a displayable attribute. This limits what a user can see if not logged in as an admin. OED does not yet have the ability to limit by login on a general basis. OED has thought about this but so far sites have mostly wanted their data public. If this is of interest to you then we should discuss what is desired and then OED can create an issue/design doc.
